11_bank_card.py

```diff
@@ -0,0 +1,124 @@
+"""
+
+Hometask 14 lambda.
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters,
+# __str_ magic method
+# do not forget about block
+
+"""
+
+import datetime
+import random
+
+
+class BankCard:
+    """Make class."""
+
+    bank = 'Best Bank'
+    card = 'VISA'
+
+    def __init__(self, name, surname, balance):
+        """Make init."""
+        self.name = name
+        self.surname = surname
+        self.balance = balance
+        self.card_num = self.card_number
+        self.exp_date = self.exp_date()
+        self.card_pin = self.card_pin()
+        self.card_cvv = self.card_cvv()
+
+    @staticmethod
+    def card_number():
+        """secret_card = f"{'*' * 12}{card[-4:]}."""
+        """card = str(random.randint(1000000000000000, 9999999999999999))
+        it's also works."""
+        card = str(random.randint(10 ** 15, (10 ** 16) - 1))
+        return f"{'*' * 12}{card[-4:]}"
+
+    @staticmethod
+    def exp_date():
+        """Find exp date."""
+        today = datetime.date.today()
+        exp_month = today.month
+        exp_year = int(str(today.year + 4)[2:])
+        return f'{exp_month:02d}/{exp_year}'
+
+    @staticmethod
+    def card_pin():
+        """Make static method."""
+        return random.randint(0000, 9999)
+
+    @staticmethod
+    def card_cvv():
+        """Find random cvv code."""
+        return random.randint(100, 999)
+
+    @classmethod
+    def get_account_info(cls):
+        """Bank account."""
+        return f'Bank is {cls.bank}'
+
+    @classmethod
+    def get_card_info(cls):
+        """Card type."""
+        return f"Card's type is {cls.card}"
+
+    @classmethod
+    def balance(cls):
+        """Show balance."""
+        return cls.balance
+
+    def transfer_money(self, balance):
+        """Make transfer money from ATM."""
+        amount_transfer = float(input('Please enter transfer amount: '))
```

Comment:
Also please do not use interactive inputs
Best practice -> use function arguments instead

Reply:
All done))